packages/components/src/components/hds/app-side-nav/list/link.hbs

```diff
@@ -0,0 +1,50 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<Hds::AppSideNav::List::Item>
```

Comment:
Not sure exactly where to comment these, but I have a few notes from testing all related to list items (I think):

1. (Bear with me, somewhat confusing language) for the standard list items (ones that aren't active/current), the active/mouse down state is the same as the resting/default state for the active/current list item, where they should follow the interactive states outlined [here](https://www.figma.com/design/noyY6dUMDYjmySpHcMjhkN/branch/WVcyo3y0R8Cf5ZatlfUkuk/HDS-Components?m=auto&node-id=64810-18282&t=Pt2sSNKzYnjO4dMl-1). Let me know any questions about this, I tried to outline it a bit in Figma.

2. The Active/Current link still has a focus state which I found somewhat strange, I think we determined that there wouldn't be interactive states for this element because it doesn't change the view/route in the UI. Additionally, the design changes somewhat when it's focused.
<img width="354" alt="Screenshot 2024-09-17 at 4 04 41 PM" src="https://github.com/user-attachments/assets/be3dfd2c-ff73-4a6e-b2b4-78b83ced12cf">

3. It looks like the focus ring on the "standard" list items is getting cut off a bit by the list item below. I think this is caused by the list items having a background color in their default state (which is the same as the background color for the container of the component. What do you think about removing that and instead having a transparent background in the resting state?
<img width="282" alt="Screenshot 2024-09-17 at 3 42 33 PM" src="https://github.com/user-attachments/assets/35fa49fb-4a14-44c8-8bf1-579ae3c215e5">

4. When the components expands from it's collapsed state, the blue border for the active/current link looks like it's stretched? It's kind of a strange visual bug, probably because the width of the list item is expanding which causes the height to decrease. I tried to capture a screen recording of it:
![CleanShot 2024-09-17 at 15 47 14](https://github.com/user-attachments/assets/a0b23ac0-3637-4fa1-9cb1-0e1f2302723a)

Reply:
Trying to go through these one by one... Btw, it may be helpful to look at the Showcase examples for these as well: https://hds-showcase-git-hds-3800-app-sidenav-component-hashicorp.vercel.app/components/app-side-nav

* For the focus state on the "active" link, that's there because it was in the original SideNAv. I'll look to remove.

Reply:
Ok for the first point, everything looks like it's mapped correctly, but when I click on a link (mouse down), it appears that that active state is is the same as the active/current link. But I wonder if this is a showcase issue because all of the links are local. Might not be an issue outside of the showcase, but attaching a screen recording just in case (excuse the jumping to a new page).

![CleanShot 2024-09-19 at 10 15 51](https://github.com/user-attachments/assets/81be6604-bc66-4d1d-b354-ed27e1abef8a)

Reply:
Pushed up a style change that seems to fix this.

Reply:
@jorytindall When you have a chance, could you please test again and let me know if it needs any more adjusting?

* [AppSideNav Showcase](https://hds-showcase-git-hds-3800-app-sidenav-component-hashicorp.vercel.app/components/app-side-nav)
* [Framed example used in context](https://hds-showcase-git-hds-3800-app-sidenav-component-hashicorp.vercel.app/layouts/app-frame/frameless/demo-full-app-frame-with-app-header-and-app-side-nav)

Reply:
Nice, yep, that seems to fix it!